src/sql/workbench/contrib/notebook/browser/models/notebookInput.ts

```diff
@@ -42,6 +42,7 @@ export class NotebookEditorModel extends EditorModel {
 	private _notebookTextFileModel: NotebookTextFileModel;
 	private readonly _onDidChangeDirty: Emitter<void> = this._register(new Emitter<void>());
 	private _lastEditFullReplacement: boolean;
+	private _isFirstKernelChange: boolean = true;
```

Comment:
For future I would put this behind a guard so it can't be set to true after it's set to false as a defensive measure.